IRCConnector.py

```diff
@@ -30,17 +37,20 @@ def run(self):
 
 
 class IRCLego(Lego):
-    def __init__(self, channel, nickname, server, baseplate, port=6667):
-        super(Lego, self).__init__(baseplate)
-        self.botThread = IRCBot(channel, nickname, server, baseplate, port)
+    def __init__(self, channel, nickname, server, baseplate, lock, port=6667, use_ssl=False):
+        super().__init__(baseplate, lock)
+        self.botThread = IRCBot(channel, nickname, server, baseplate, port, use_ssl)
+        self.channel = channel
 
     def on_start(self):
         self.botThread.start()
 
     def listening_for(self, message):
-        # print('asking irc listener if it wants that')
-        # return (str(self) != str(message['metadata']['source']))
-        return True
+        return (str(self.botThread) != str(message['metadata']['source']))
```

Comment:
We're checking whether the message came from this Lego. If it didn't, we're relaying it. This relays all messages without echoing.